core/src/main/kotlin/org/rewedigital/katana/dsl/Dsl.kt

```diff
@@ -79,6 +79,19 @@ inline fun <reified T> ModuleBindingContext.eagerSingleton(
         provider = DefaultProvider(body)
     )
 
+/**
+ * Declares an alias of type [A] with optional [name] of an already existing binding of type [T] with optional name
+ * [originalName]. An alias is accomplished by and nothing else than a simple [factory] binding.
+ *
+ * A binding like `alias<MyClass, MyInterface>()` should be conceived as "alias MyClass as MyInterface".
+ */
+inline fun <reified T : A, reified A> ModuleBindingContext.alias(
+    originalName: String? = null,
```

Comment:
Shouldn't this be `Any?`? Also, I would have maybe inversed `T` and `A`, since `T` is what is being declared, like everywhere else, `A` is just the added reference to the binding being aliased... Also, I wonder if there's any sense to have the name here in the first place, since anyways named bindings ignore type for resolution...? Unless one wants to use a named binding to declare one without a name... but that might be confusing.

Reply:
I reversed the types so that

```
alias<MyClass, MyInterface>()
```

can be read as "alias MyClass as MyInterface". I think the other way around might be confusing? Also the UNIX `ln` command for instance expects the target as the first argument and the link name as the second argument. Maybe we want to adhere to that?

Reply:
Why should it be `Any`? Shouldn't the alias be assignable to the original type?

Reply:
But then, `singleton<MyInterface> { MyClass() }` reads the opposite way... I wonder what's more confusing, changing from the style used in the library or not using Kotlin's `as` or UNIX `ln` syntax... I tend to think a user being in the Context of the library being used... but whatever way I guess the user will get used to since, `T : A` won't let them do the wrong thing.

Reply:
I think (optional) names should be part of an alias as it should be possible to alias a named binding under an class/interface and vice versa.

```
singelton(name = "my singleton") { MyClass() }

// Alias named binding "my singleton" with type MyClass as MyInterface (w/o a name)
alias<MyClass, MyInterface>(originalName = "my singleton")
```

Reply:
If so, it should be `factory<A>() { get<T>(name = originalName) }` not `factory<A>(name = name) { get<T>(name = originalName) }`, no?

Reply:
> If so, it should be `factory<A>() { get<T>(name = originalName) }` not `factory<A>(name = name) { get<T>(name = originalName) }`, no?

Yes, but it should *imho* also be possible to specify a new name for the alias. Since names are purely optional in both cases (for the original item and the alias) I think it's okay to offer both possibilities.

Reply:
> But then, `singleton<MyInterface> { MyClass() }` reads the opposite way... I wonder what's more confusing, changing from the style used in the library or not using Kotlin's `as` or UNIX `ln` syntax... I tend to think a user being in the context of the library being used... but whatever way I guess the user will get used to since, `T : A` won't let them do the wrong thing.

I will reverse the type arguments to be in line with the rest of the Katana syntax

Reply:
Makes sense. I still feel a bit uncomfortable about the order, but I'll get used to it :smile:. What about `alias` for `set { }`?

Reply:
> What about `alias` for `set { }`?

I thought about this but *imho* it doesn't make sense here. Any binding inside `set` is locked to the type of the set. Also these bindings cannot have names. So there is just a single type inside `set` anyway.

Reply:
What about a singleton declared outside of a set? All we would need is to refer to that instance inside the set... (that's our use case).

Reply:
> What about a singleton declared outside of a set? All we would need is to refer to that instance inside the set... (that's our use case).

I just added `get()` "binding" to `set`. Please have a look.